packages/repeater/src/request-runner/Request.ts

```diff
@@ -0,0 +1,82 @@
+import { URL } from 'url';
+
+export interface RequestOptions {
+  method?: string;
+  url: string;
+  headers: Record<string, string | string[]>;
+  body?: string;
+  correlationIdRegex?: string | RegExp;
+}
+
+export class Request {
+  public readonly url: string;
+  public readonly body?: string;
+  public readonly correlationIdRegex?: RegExp;
+
+  private readonly _method?: string;
+
+  get method(): string | undefined {
+    return this._method;
+  }
+
+  private _headers: Record<string, string | string[]>;
+
+  get headers(): Record<string, string | string[]> {
```

Comment:
To prevent it being modified outside

```suggestion
  get headers(): Readonly<Record<string, string | string[]>> {
```